lib/tasks/keygen/permissions.rake

```diff
@@ -3,47 +3,124 @@
 namespace :keygen do
   desc 'Tasks for managing permissions'
   namespace :permissions do
+    task :add, %i[type] => %i[environment] do |_, args|
+      batch_size = ENV.fetch('BATCH_SIZE') { 1_000 }.to_i
+      batch_wait = ENV.fetch('BATCH_WAIT') { 0.1 }.to_f
+
+      model = args[:type].to_s.classify.safe_constantize
+
+      # Split args up into ID and permission buckets
+      ids,
+      permissions,
+      * = args.extras.chunk   { Permission::ALL_PERMISSIONS.include?(_1) }
```

Comment:
This needs to be sorted, no? That way we don't mix up the ID and permission arrays.